djstripe/migrations/0012_transfer_to_destination_fk.py

```diff
@@ -0,0 +1,36 @@
+# Generated by Django 3.2.4 on 2021-06-22 09:48
+from django.db import migrations
+
+
+def transfer_to_destination(apps, schema_editor):
```

Comment:
Right so I missed this file in the first review. Imperative python migrations are pretty bad, only to use for the edgiest of edge cases. At a glance, I believe this data migration can be done in SQL, so I'm not comfortable merging it in as it is.

Is there anything in this PR that doesn't require this migration? I'd like to land whatever we can and keep working on this one in the mean time.

Reply:
My understanding of this migration is that you have a CharField that contains an ID, and want to turn it into a foreign key to an account. IMO you do not need a data migration to do that: The data is the same.

Scenarios to address:

1. The column needs to be renamed, as django appends `_id` to it. This can be solved by adding a `db_field=` to the foreign key.
2. StripeForeignKeys have a different `to_field` that depends on a setting. You want to enforce the `to_field` so that it is always the stripe ID, never the `djstripe_id` regardless of setting. This is OK because we want to deprecate djstripe_id altogether.
3. No deprecation period right now, the char field becomes a foreign key. This needs to be pointed out in the 2.6.0 changelog as a breaking change; make sure to add it (a `2_6_0.md` file might have to be created, see git log on the other ones if you're unsure about the format).

Did I miss anything?

Reply:
Let me get back to you after going through the changes once again.

Reply:
Regarding the Migration:

1. Yes, that is what I was trying to do. I think you got it all.
2. I think there could be issues with data being out of sync in the case of an existing model existing as a `StripeIdField` on another model.   For example, not all models have their own `Stripe Webhooks` so their updates need to be implemented in some other webhook or they exist as `FKs` on a model with an "exposed" Stripe Webhook in which case the code automatically fetches and syncs them along with the model they are an `FK` to. This, however, would not happen in general and one would need to audit the specific webhooks, `_attach hooks` etc, which to me seems like more work in the long run. Totally doable though.
2. Yes, the data migration can be done in SQL. 

So just to be clear you're okay with doing this?

Reply:
I'm ok with doing it in SQL instead of python yeah.

Reply:
We should also ensure there's a backwards migration, btw.

Reply:
Ok I will do that. In the current code, as long as the field is nullable backward migration should work. I added a `noop` operation since the backward migration would essentially delete the column with data anyway because  apart from the data migration we simply `add, alter, and remove` fields. I will test more especially with `null` values as that can sometimes be tricky to work with.